src/renderer/main/panels/preferences/MeltSelect.svelte

```diff
@@ -16,51 +22,16 @@
       fitViewport: true,
       sameWidth: true,
     },
-    defaultSelected: { value: target.value, label: target.title },
+    defaultSelected: getDefaultSelected(),
   });
 
-  $: {
-    if ($selected) {
-      handleSelectionChange();
-    }
-  }
-
-  function handleSelectionChange() {
-    if (typeof $selected === "undefined") {
-      if (typeof target !== "undefined") {
-        selected.set(() => {
-          const s = { value: undefined, label: undefined };
-          const item = options.find((e) => {
-            return e.value === target;
-          });
-          if (typeof item !== "undefined") {
-            s.value = target;
-            s.label = item.title;
-            oldTarget = target;
-            console.log(s);
-            return s;
-          }
-        });
-      }
-    } else {
-      if (target !== oldTarget) {
-        selected.update((s) => {
-          const item = options.find((e) => {
-            return e.value === target;
-          });
-          if (typeof item !== "undefined") {
-            s.value = target;
-            s.label = item.title;
-            oldTarget = target;
-            return s;
-          }
-        });
-      }
+  $: handleSelectionChange($selected);
 
-      if (target !== $selected.value) {
-        oldTarget = target = $selected.value;
-      }
+  function handleSelectionChange(selected: { label: string; value: any }) {
+    if (selected === target) {
+      return;
```

Comment:
Use case:
NEVER update target if no actual selection change happened
If target is updated from the outside (correctly) change selection, (target should not change twice, in this case again from the inside)
If target is changed from the inside, propagate change to the outside.